templates/go/cloudevents/handle.go

```diff
@@ -2,24 +2,57 @@ package function
 
 import (
 	"context"
+	"encoding/json"
+	"errors"
 	"fmt"
+	"strings"
 
-	event "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/event"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
 )
 
-// Handle an event.
-func Handle(ctx context.Context, event event.Event) error {
+type Output struct {
+	Output    string `json:"output"`
+	Input     string `json:"input"`
+	Operation string `json:"operation"`
+}
 
-	/*
-	 * YOUR CODE HERE
-	 *
-	 * Try running `go test`.  Add more test as you code in `handle_test.go`.
-	 */
+type Input struct {
+	Input string `json:"input"`
+}
 
-	// Example implementation:
-	fmt.Printf("%v\n", event) // print the received event to standard output
+func uppercase(event cloudevents.Event) (*event.Event, error) {
+	input := Input{}
+	err := json.Unmarshal(event.Data(), &input)
+	if err != nil {
+		fmt.Printf("%v\n", err)
+		return nil, err
+	}
+	fmt.Printf("%v\n", input)
+	outputEvent := cloudevents.NewEvent()
+	outputEvent.SetSource("http://example.com/uppercase")
+	outputEvent.SetType("UpperCasedEvent")
+	output := Output{}
+	output.Input = input.Input
+	output.Output = strings.ToUpper(input.Input)
+	output.Operation = event.Subject()
+	outputEvent.SetData(cloudevents.ApplicationJSON, &output)
+
+	fmt.Printf("Outgoing Event: %v\n", outputEvent)
 
-	return nil
+	return &outputEvent, nil
+}
+
+// Handle an event.
+func Handle(ctx context.Context, event cloudevents.Event) (*event.Event, error) {
+
+	// Example implementation:
+	fmt.Printf("Incoming Event: %v\n", event) // print the received event to standard output
+	if event.Type() == "uppercase" {
+		return uppercase(event)
```

Comment:
Builtin Functions should echo their input.  Any builtin Functions which do not have this as their one and only behavior should be updated to match.  This will allow us to expand our integration and E2E tests to run for each language.

The builtin Functions are not app examples, they are a default chosen function signature, an example test and an echo.

For examples, this code would be a great addition to the [Function Examples ](https://github.com/boson-project/templates) templates repository!

Reply:
@lkingland thanks a lot for the feedback.. and thanks a lot for all the pointers.. I totally agree.. let's have examples but an echo function that returns something and that can work when you build a freshly created template. 
I will move this code to the examples, so also we can have the same example for all the languages.. that will help with a bit more complex setup than just an echo for people looking for more advanced handling of events or having multiple functions in a single project

Reply:
I don't want to derail this conversation, but @lkingland your comment struck me.

> Builtin Functions should probably have no actual implementation other than to echo their input.

I agree, but have noticed that the invoke command can be a little surprising if you are expecting to see the function response. We may want to think about providing a way to inspect the response as you can with `curl`, displaying any response data as well as an option to display headers.

Reply:
That is correct, the functions should echo what they received:
```
func Handle(ctx context.Context, event cloudevents.Event) (*event.Event, error) {
  /*
   * YOUR CODE HERE
   *
   * Try running `go test`.  Add more test as you code in `handle_test.go`.
   */
  return &event, nil
}
```